src/Http/Routing/src/EndpointSelectorContext.cs

```diff
@@ -3,68 +3,52 @@
 
 using System;
 using Microsoft.AspNetCore.Http;
-using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Http.Endpoints;
 
 namespace Microsoft.AspNetCore.Routing
 {
-    public sealed class EndpointSelectorContext : IEndpointFeature, IRouteValuesFeature, IRoutingFeature
+    public struct EndpointSelectorContext
```

Comment:
This will break anyone who has written and distributed an endpoint selector. Can you think of any other side effects @rynowak?

Reply:
I’m like 87% sure we’re ok breaking that.

Reply:
I think we should break it. There are a few other things we should think about breaking.

Reply:
I think that set of people is probably just @commonsensesoftware - either way we'll announcement it.

Reply:
Marked the PR as breaking change

Reply:
Thanks for the heads up. I looked at this change and it looks to be a non-issue for me. Best as I can tell, this will be 100% source compatible with what I already have. The only write scenario I have is explicitly setting the endpoint as a short-circuit for invalid client requests (ex: duplicate API versions, no API version, etc). It appears this change doesn't alter that behavior.

What will this mean for route constraints - anything? I believe today they are still called in _Endpoint Routing_, but supply `null` for the **IRouter** parameter. That doesn't affect me, but I was wondering if I should expect a possible change there as well. I only use a route constraint for URL segment versioning so I don't need to perform _magic string_ parsing on the URL to find where the version value is in the path.

Reply:
There's no change for `IRouteContraint`, they are used by both routing systems.